molo/core/wagtail_hooks.py

```diff
@@ -80,10 +80,23 @@ def show_reactionquestions_response_for_users_have_access(request, menu_items):
 def show_main_language_only(parent_page, pages, request):
     main_language = Languages.for_site(request.site).languages.filter(
         is_main_language=True).first()
-    specific_pages = [page.specific for page in pages]
+    if isinstance(pages, QuerySet):
+        specific_pages = pages
+        for page in pages:
+            if not page.specific:
+                specific_pages = specific_pages.exclude(pk=page.pk)
+    else:
+        specific_pages = [page.specific for page in pages]
     if pages and main_language and parent_page.depth > 2:
-        new_pages = [page for page in specific_pages
-                     if page.language and page.language.pk == main_language.pk]
+        if isinstance(specific_pages, QuerySet):
+            new_pages = specific_pages
+            for page in specific_pages:
+                if not page.language and page.language.pk == main_language.pk:
```

Comment:
I think this is wrong. the previous one is "if x and y" so this needs to be "if not x or not y"
Alternatively you could do "if not (x and y)"

Reply:
fixed